servo/connectors/kubernetes.py

```diff
@@ -2087,9 +2144,8 @@ async def handle_error(self, error: Exception, mode: "FailureMode") -> bool:
                 must filter failure modes before calling the superclass implementation.
         """
         if mode == FailureMode.crash:
-            raise RuntimeError(
-                "an unrecoverable failure occurred while interacting with Kubernetes"
-            ) from error
+            self.logger.error(f"an unrecoverable failure occurred while interacting with Kubernetes: {error.__class__.__name__} - {str(error)}")
+            raise error
```

Comment:
`RuntimeError` was kind of a lazy solution here but my intent was to use exception chaining so that you would have a known surface area to trap with an except block since its hard to know what's gonna bubble up here

Reply:
Sounds reasonable, we'd just need to add the logic to unwrap chained exceptions since, currently, the top-most exception in the chain is the only one that I saw show up on the Web UI